cpp/arcticdb/version/version_core.cpp

```diff
@@ -821,16 +821,26 @@ void read_incompletes_to_pipeline(
     pipeline_context->total_rows_ = pipeline_context->calc_rows();
 }
 
-void check_incompletes_index_ranges_dont_overlap(const std::shared_ptr<PipelineContext>& pipeline_context) {
-    // Does nothing if the symbol is not timestamp-indexed
-    // Checks both that the index ranges of incomplete segments do not overlap with one another, and that the earliest
-    // timestamp in an incomplete segment is greater than the latest timestamp existing in the symbol in the case of a
-    // parallel append
+void check_incompletes_index_ranges_dont_overlap(const std::shared_ptr<PipelineContext>& pipeline_context,
```

Comment:
nit (style): Passing shared_ptr by const ref is [not advised](https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/).

Reply:
Fair, it is an extremely common pattern in our codebase though

Reply:
PipelineContext is going to get refactored out of existence anyway